src/main/java/com/powsybl/openloadflow/sensi/DcSensitivityAnalysis.java

```diff
@@ -697,11 +888,12 @@ private void detectConnectivityLoss(LfNetwork lfNetwork, DenseMatrix states, Lis
                     factorsStates, contingenciesStates, Collections.emptyList());
 
             // connectivity analysis by connectivity
-            // we will index contingencies by a list of branch associations that breaks connectivity
+            // we will index contingencies by a list of branch that may breaks connectivity
             // for example, if in the network, loosing line L1 breaks connectivity, and loosing L2 and L3 together breaks connectivity,
-            // the associations would be: [{L1}, {L2, L3}]
+            // the index would be: {L1, L2, L3}
+            // todo: There may be a better way to group the contingencies that will have the same connectivities afterwards
             Collection<Contingency> nonLosingConnectivityContingencies = new LinkedList<>();
-            Map<List<Set<ComputedContingencyElement>>, List<Contingency>> contingenciesByGroupOfElementsBreakingConnectivity = new HashMap<>();
+            Map<Set<ComputedContingencyElement>, List<Contingency>> contingenciesByGroupOfElementsBreakingConnectivity = new HashMap<>();
```

Comment:
I'm a bit confused with the map here, as we don't need the indexing. Just a list of pairs would be ok (or a list of other custom objects). But anyway that doesn't come from this PR, it was already like that before.